frontend/src/components/data-table/pagination.tsx

```diff
@@ -27,10 +28,19 @@ export const DataTablePagination = <TData,>({
   totalColumns,
 }: DataTablePaginationProps<TData>) => {
   const renderTotal = () => {
-    const selected = Object.keys(table.getState().rowSelection).length;
-    const isAllPageSelected = table.getIsAllPageRowsSelected();
+    const { rowSelection, cellSelection } = table.getState();
+    let selected = Object.keys(rowSelection).length;
+    let isAllPageSelected = table.getIsAllPageRowsSelected();
     const numRows = table.getRowCount();
-    const isAllSelected = selected === numRows;
+    let isAllSelected = selected === numRows;
+
+    const isCellSelection =
+      selection === "single-cell" || selection === "multi-cell";
+    if (isCellSelection) {
+      selected = cellSelection.length;
+      isAllPageSelected = table.getIsAllPageCellsSelected();
+      isAllSelected = table.getIsAllCellsSelected();
```

Comment:
i think we don't need to support `isAllPageSelected` and `isAllSelected`, so this can be `false`.

we also may want to remove the top checkbox from the checkbox column (if not already)